### PR TITLE
Update BroadcastExecutor.java for working links

### PR DESCRIPTION
**Pull request by GigiaJ**
The changes I made are solely to the BroadcastExecutor, I added a RegEx that ran through checks after the message was confirmed to have a link (http or https). If it had them, three regEx were ran. One for all of the message before the link, one for the link itself, and one for all the message after the link. Basically it cuts the message into parts and reforms them. If it doesn't have a link(https or http) then it simply runs the old command. I'd appreciate any criticism as I'm just learning Java, but I would love to assist you in this. This fix may also be applicable to normal chat also; I just haven't tinkered with it quite yet.

### CODE REVIEW

File: src/main/java/io/github/hsyyid/essentialcmds/cmdexecutors/BroadcastExecutor.java [hunk 1]
```diff
@@ -56,7 +133,7 @@ public void executeAsync(CommandSource src, CommandContext ctx) {
 	@Nonnull
 	@Override
 	public CommandSpec getSpec() {
-		return CommandSpec.builder().description(Text.of("Broadcast Command")).permission("essentialcmds.broadcast.use")
-				.arguments(GenericArguments.remainingJoinedStrings(Text.of("message"))).executor(this).build();
+	return CommandSpec.builder().description(Text.of("Broadcast Command")).permission("essentialcmds.broadcast.use")
+	.arguments(GenericArguments.remainingJoinedStrings(Text.of("message"))).executor(this).build();
```
Comment (hsyyid) [inline src/main/java/io/github/hsyyid/essentialcmds/cmdexecutors/BroadcastExecutor.java]: What'd you do to formatting? Why'd you add a space after this?

File: src/main/java/io/github/hsyyid/essentialcmds/cmdexecutors/BroadcastExecutor.java [hunk 2]
```diff
@@ -25,28 +25,105 @@
...
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.spongepowered.api.text.action.TextActions;
 import javax.annotation.Nonnull;
 
+
```
Comment (hsyyid) [inline src/main/java/io/github/hsyyid/essentialcmds/cmdexecutors/BroadcastExecutor.java:45]: Why a new line??

File: src/main/java/io/github/hsyyid/essentialcmds/cmdexecutors/BroadcastExecutor.java [hunk 3]
```diff
@@ -25,28 +25,105 @@
 package io.github.hsyyid.essentialcmds.cmdexecutors;
 
 import io.github.hsyyid.essentialcmds.internal.AsyncCommandExecutorBase;
+
```
Comment (hsyyid) [inline src/main/java/io/github/hsyyid/essentialcmds/cmdexecutors/BroadcastExecutor.java]: Why a new line??

File: src/main/java/io/github/hsyyid/essentialcmds/cmdexecutors/BroadcastExecutor.java [hunk 4]
```diff
@@ -25,28 +25,105 @@
...
 import org.spongepowered.api.command.CommandSource;
 import org.spongepowered.api.command.args.CommandContext;
 import org.spongepowered.api.command.args.GenericArguments;
 import org.spongepowered.api.command.spec.CommandSpec;
 import org.spongepowered.api.text.Text;
+
```
Comment (hsyyid) [inline src/main/java/io/github/hsyyid/essentialcmds/cmdexecutors/BroadcastExecutor.java]: Why a new line?

File: src/main/java/io/github/hsyyid/essentialcmds/cmdexecutors/BroadcastExecutor.java [hunk 5]
```diff
@@ -25,28 +25,105 @@
...
 import org.spongepowered.api.text.format.TextColors;
 import org.spongepowered.api.text.serializer.TextSerializers;
 
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
```
Comment (hsyyid) [inline src/main/java/io/github/hsyyid/essentialcmds/cmdexecutors/BroadcastExecutor.java]: Why a new line?

File: src/main/java/io/github/hsyyid/essentialcmds/cmdexecutors/BroadcastExecutor.java [hunk 6]
```diff
@@ -25,28 +25,105 @@
...
-		Text msg = TextSerializers.formattingCode('&').deserialize(message);
-		Text broadcast = Text.of(TextColors.DARK_GRAY, "[", TextColors.DARK_RED, "Broadcast", TextColors.DARK_GRAY, "]", TextColors.GREEN, " ");
-		Text finalBroadcast = Text.builder().append(broadcast).append(msg).build();
-		MessageChannel.TO_ALL.send(finalBroadcast);
+
+    @Override
+    public void executeAsync(CommandSource src, CommandContext ctx) {
+	
+    String message = ctx.<String> getOne("message").get();
+	Text msg = TextSerializers.formattingCode('&').deserialize(message);
```
Comment (hsyyid) [inline src/main/java/io/github/hsyyid/essentialcmds/cmdexecutors/BroadcastExecutor.java]: This stuff is not aligned correctly..

File: src/main/java/io/github/hsyyid/essentialcmds/cmdexecutors/BroadcastExecutor.java [hunk 7]
```diff
@@ -25,28 +25,105 @@
...
+			String preurlline = preurl.replaceAll("(((https?|ftp|gopher|telnet|file):((//)|(\\\\))+[\\w\\d:#@%/;$()~_?\\+-=\\\\\\.&]*).*?)(?=\\w).*","");
+			String posturlline = posturl.replaceAll("((https?|ftp|gopher|telnet|file|Unsure|http):((//)|(\\\\))+[\\w\\d:#@%/;$()~_?\\+-=\\\\\\.&]*)","");
+			Text Postmessage= Text.builder(posturlline).build();
+			Text Premessage = Text.builder(preurlline).build();
+			Text linkmessage = Text.builder(url).onClick(TextActions.openUrl(new URL(url))).build();
+			Text newmessage = Text.builder().append(Premessage).append(linkmessage).append(Postmessage).build();
+			Text Glory = Text.builder().append(broadcast).append(newmessage).build();
+			MessageChannel.TO_ALL.send(Glory);}
+		catch (MalformedURLException e) {
+		// TODO Auto-generated catch block
```
Comment (hsyyid) [inline src/main/java/io/github/hsyyid/essentialcmds/cmdexecutors/BroadcastExecutor.java]: Please get rid of this todo

**Comment by hsyyid** [comment]
Almost all the formatting is messed up... Please fix this and then I'll test it out.

**Comment by GigiaJ** [comment]
Sorry I'm extremely bad with comparing things. I appreciate you taking the time to look at it and point out the faults for me.

File: src/main/java/io/github/hsyyid/essentialcmds/cmdexecutors/BroadcastExecutor.java [hunk 8]
```diff
@@ -1,7 +1,7 @@
 /*
  * This file is part of EssentialCmds, licensed under the MIT License (MIT).
  *
- * Copyright (c) 2015 - 2016 HassanS6000
```
Comment (kinggoesgaming) [inline src/main/java/io/github/hsyyid/essentialcmds/cmdexecutors/BroadcastExecutor.java]: mate what? this 2016 now not 2015

Comment (GigiaJ) [inline src/main/java/io/github/hsyyid/essentialcmds/cmdexecutors/BroadcastExecutor.java]: It wouldn't build right without that 2015. It was originally 2015 until I changed it to 2016 then reverted it.

**Pull request merged**